### PR TITLE
Refactor account

### LINKED ISSUE #435: Refactor Account package
- [ ] Create Json package.json
- [ ] Add account package to Workspace
- [ ] Remove Frontend related Code
- [ ] Remove Backend related Code
- [ ] Remove Stroage related Code

### PR DESCRIPTION
**Pull request by AlexNi245**
closes #435 #436

### CODE REVIEW

File: packages/lib/src/account/package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          "require": "./dist.backend/index.js"
+      }
+  },
+  "dependencies": {
+    "@types/libsodium-wrappers": "^0.7.10",
+    "ajv": "^8.11.0",
+    "axios": "^0.25.0",
+    "ethers": "^5.5.4",
+    "googleapis": "^100.0.0",
+    "libsodium-wrappers": "^0.7.10",
```
Comment (hai-ko) [inline packages/lib/src/account/package.json:24]: Do we need this dependency outside of crypto?

File: packages/lib/src/account/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          "import": "./dist/index.js",
+          "require": "./dist/index.js"
+      },
+      "./dist.backend": {
+          "import": "./dist.backend/index.js",
+          "require": "./dist.backend/index.js"
+      }
+  },
+  "dependencies": {
+    "@types/libsodium-wrappers": "^0.7.10",
```
Comment (hai-ko) [inline packages/lib/src/account/package.json:19]: Could be ad dev dependency

File: packages/lib/src/account/src/profileKeys/createProfileKeys.test.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import { ethers } from 'ethers';
-import { createStorageKey, getStorageKeyCreationMessage } from '../../crypto';
+import {
+    createStorageKey,
+    getStorageKeyCreationMessage,
+} from '../../../crypto/src';
```
Comment (hai-ko) [inline packages/lib/src/account/src/profileKeys/createProfileKeys.test.ts:5]: without `src`

File: packages/lib/src/account/src/profileKeys/createProfileKeys.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { createKeyPair, createSigningKeyPair } from '../../crypto';
+import { createKeyPair, createSigningKeyPair } from '../../../crypto/src';
```
Comment (hai-ko) [inline packages/lib/src/account/src/profileKeys/createProfileKeys.ts:1]: without `src`

File: packages/lib/src/delivery/Delivery.ts [hunk 5]
```diff
@@ -4,11 +4,11 @@ import axios, {
     AxiosRequestConfig,
     AxiosResponse,
 } from 'axios';
-import { GetResource, UserProfile } from '../account/Account';
-import { IpfsResolver } from '../account/profileResolver/IpfsResolver';
-import { JsonResolver } from '../account/profileResolver/JsonResolver';
-import { LinkResolver } from '../account/profileResolver/LinkResolver';
-import { ProfileResolver } from '../account/profileResolver/ProfileResolver';
+import { GetResource, UserProfile } from '../account/src/Account';
```
Comment (hai-ko) [inline packages/lib/src/delivery/Delivery.ts:7]: without `src`

File: packages/lib/src/delivery/Messages.test.ts [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import { BigNumber, ethers } from 'ethers';
-import { normalizeEnsName, UserProfile } from '../account/Account';
-import { decryptAsymmetric, encryptAsymmetric } from '../crypto';
-import { DeliveryInformation, EncryptionEnvelop } from '../messaging/Envelop';
-import { stringify } from '../shared/stringify';
+import { testData } from '../../../../test-data/encrypted-envelops.test';
+import { normalizeEnsName, UserProfile } from '../account/src/Account';
+import { decryptAsymmetric } from '../crypto/src';
```
Comment (hai-ko) [inline packages/lib/src/delivery/Messages.test.ts:4]: without `src`

File: packages/lib/src/delivery/Messages.ts [hunk 7]
```diff
@@ -1,16 +1,16 @@
 import { ethers } from 'ethers';
 import stringify from 'safe-stable-stringify';
-import { normalizeEnsName } from '../account';
+import { normalizeEnsName } from '../account/src';
```
Comment (hai-ko) [inline packages/lib/src/delivery/Messages.ts:3]: without `src`

File: packages/lib/src/delivery/Messages.ts [hunk 8]
```diff
@@ -1,16 +1,16 @@
...
 
 import {
     decryptAsymmetric,
     encryptAsymmetric,
     EncryptedPayload,
     KeyPair,
     sign,
-} from '../crypto';
+} from '../crypto/src';
```
Comment (hai-ko) [inline packages/lib/src/delivery/Messages.ts:11]: without `src`

**Review by hai-ko — CHANGES_REQUESTED** [review]

File: packages/lib/src/account/package.json (hunk 1 above)
Comment (AlexNi245) [reply packages/lib/src/account/package.json:24]: I've not reviewed the particular dependencies just copied the from the parent package.json

File: packages/lib/src/account/src/Account.ts [hunk 9]
```diff
@@ -1,27 +1,17 @@
...
-    getConversationId,
-    UserDB,
-} from '../storage/Storage';
-import { Connection } from '../web3-provider/Web3Provider';
+} from '../../external-apis/InjectedWeb3API';
+import { log } from '../../shared/src/log';
+import { sha256 } from '../../shared/src/sha256';
+import { stringify } from '../../shared/src/stringify';
+import { Connection } from '../../web3-provider/Web3Provider';
+import { KeyPair } from './../../crypto/src';
```
[the remark this answers is not in the document]
Comment (AlexNi245) [reply packages/lib/src/account/src/Account.ts:14]: doesn't work without the src because the files are located in the src dir

**Review by hai-ko — APPROVED** [review]

**Pull request closed without merging**